### PR TITLE
Work around Chrome Autofill issue

### LINKED ISSUE #747: [BUG] Autofill not working with Chromium based browsers
**Describe the bug**
Autofill does not work in browser (tested on Brave(main) and also Chrome). It works as expected in other types of apps (for example Netflix).

**To Reproduce**
1. Open browser
2. Go to a website
3. Select the username or password field
4. No prompt to select password

**Expected behavior**
When a password field is selected in browser the prompt for picking a username/password shall be triggered

**Device information (please complete the following information):**
 - Device: Oneplus 7T Pro
 - OS: ParanoidAndroid Quartz 2 (Android 10)
 - App version: 1.7.1

### PR DESCRIPTION
**Pull request by fmeum**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description

Work around Chrome issue 2 described below by adding an `AccessibilityService` that listens to window changes in Chrome and forces the accessibility virtual view structure to be populated.

## :bulb: Motivation and Context

I have identified at least two distinct issues with Autofill in Chrome:

1. Due to an Android bug introduced [with Android 10](https://cs.android.com/android/_/android/platform/frameworks/base/+/a7de402a17621d9af66282366605304088a46850), Autofill breaks if a view that is labeled not important for Autofill somehow (this part appears to be non-deterministic and vary between phones) triggers an Autofill request. I don't think that we can work around it, but there is the [option for a simple workaround on the Chrome side](https://chromium-review.googlesource.com/c/chromium/src/+/2275450) that I will lobby for. You can try to reproduce this bug with [this reproducer APK](https://drive.google.com/file/d/1WhlRjH8F1C4F0uNOhFU6s1Cmjb3JMr-y/view?usp=sharing).

2. Chrome takes a lazy approach to populating sccessibility info as documented [here](https://source.chromium.org/chromium/chromium/src/+/master:docs/accessibility/android.md;drc=0712ad54c423827d7f3686132016a2531c103c24), which apparently leads to accessibility events not being propagated to the Autofill accessibility compatibility bridge. You can reproduce this in Chrome by browsing to `https://github.com/login` in two different ways from the new tab page: either tap a shortcut or enter the URL in the omnibox. The former should lead to Autofill being triggered on the login page, the latter not (or at least not always). 

Might fix #747.

## :green_heart: How did you test it?
Autofill has become much more reliable with the accessibility service enabled.

## :pencil: Checklist

- [ ] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [ ] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :crystal_ball: Next steps

We should enable this service and see whether it improves Autofill (or degrades Chrome performance) on as many devices running Android 9+ as possible.

### CODE REVIEW

**Comment by msfjarvis** [comment]
I don't suppose this is going to make any difference for the first case?

**Comment by fmeum** [comment]
> I don't suppose this is going to make any difference for the first case?

I don't think so, the bug triggers well before our Autofill service is invoked and has nothing to do with the accessibility virtual view structure. This has to be worked around in Chrome, unfortunately.

**Comment by fmeum** [comment]
I pushed a new version that debounces view node population on page change events and only populates nodes synchronously on events triggered by users. Whereas Chrome's performance degraded dramatically previously, with this change I am not noticing a difference anymore.

**Comment by fmeum** [comment]
I filed the first issue as an Android bug: https://issuetracker.google.com/issues/160665161
If we are extremely lucky, this might get fixed during the Android 11 Beta, but I doubt it.

**Review by msfjarvis — DISMISSED** [review]
Seems to make things better. Gonna need at least a changelog entry and probably an onboarding flow.

**Comment by fmeum** [comment]
This has now been integrated into the settings but can also be enabled manually if the Autofill service is already enabled. I hope that this covers all possible states, but the logic in UserPreferences is necessarily rather complex.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -275,6 +275,16 @@
...
     <string name="oreo_autofill_enable_dialog_installed_browsers">Autofill support with installed browsers:</string>
+    <string name="oreo_autofill_chrome_compat_fix_summary">Make Autofill more reliable in Chrome</string>
+    <string name="oreo_autofill_chrome_compat_fix_description">This accessibility service makes
+        Autofill work more reliably in Chrome. It can only be activated if you are already using
+        Password Store as your Autofill service.\n\nThis service is only active while you are
+        using Chrome. It does not access any data or take any actions on your behalf, but forces
+        Chrome to properly forward user interactions to the Password Store Autofill
+        service.\n\nChrome\'s performance should not be noticeably affected. If you are experiencing
+        any problems with this service, please create an issue at
+        https://msfjarvis.dev/ga.
```
Comment (msfjarvis) [inline app/src/main/res/values/strings.xml]:
```suggestion
        https://msfjarvis.dev/aps.
```

Probably easier for end users.

**Review by msfjarvis — DISMISSED** [review]

**Comment by fmeum** [comment]
Forgot to commit the reformat result.

**Review by msfjarvis — DISMISSED** [review]

**Review by msfjarvis — APPROVED** [review]

**Comment by msfjarvis** [comment]
Does Actions need a virgin sacrifice to allocate resources for our builds

**Pull request merged**